SlevomatCodingStandard/Sniffs/PHP/ProperTypeCastSniff.php

```diff
@@ -0,0 +1,94 @@
+<?php declare(strict_types = 1);
+
+namespace SlevomatCodingStandard\Sniffs\PHP;
+
+use SlevomatCodingStandard\Helpers\TokenHelper;
+
+class ProperTypeCastSniff implements \PHP_CodeSniffer\Sniffs\Sniff
+{
+
+	public const CODE_FORBIDDEN_CAST_USED = 'FobriddenCastUsed';
+	public const CODE_INVALID_CAST_USED = 'IncorrectCastUsed';
+
+	private const INVALID_CASTS = [
+		'binary' => null,
+		'boolean' => 'bool',
+		'double' => 'float',
+		'integer' => 'int',
+		'real' => 'float',
+		'unset' => null,
+	];
+
+	/**
+	 * @return mixed[]
+	 */
+	public function register(): array
+	{
+		return [
+			T_STRING_CAST,
+			T_BOOL_CAST,
+			T_DOUBLE_CAST,
+			T_INT_CAST,
+			T_UNSET_CAST,
+		];
+	}
+
+	/**
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint
+	 * @param \PHP_CodeSniffer\Files\File $phpcsFile
+	 * @param int $pointer
+	 */
+	public function process(\PHP_CodeSniffer\Files\File $phpcsFile, $pointer): void
+	{
+		$tokens = $phpcsFile->getTokens();
+		$cast = $tokens[$pointer]['content'];
+
+		if (!preg_match('~^\(\s*(\S+)\s*\)\z~i', $cast, $m)) {
+			return;
+		}
+
+		[, $castName] = $m;
```

Comment:
`$castName = $matches[1];` is better, I think.

Reply:
That's what I had originally, but found list syntax more verbose. Whatever.